src/components/language-switcher.js

```diff
@@ -6,29 +6,26 @@ import "../styles/LanguageSwitcher.css";
 
 function LanguageSwitcher({ slug }) {
   const { i18n } = useTranslation();
-  const currentLang = i18n.language;
 
   const languages = [
     { code: "en", label: "English" },
     { code: "es", label: "Spanish" },
   ];
 
   const handleChange = ({ target }) => {
-    i18n.changeLanguage(target.value);
+    const newLang = target.value;
+    i18n.changeLanguage(newLang);
+    navigate(`/${newLang}/${slug}`);
```

Comment:
If you move this to the handle change, does it still navigate to a new page if the language is changed not using the dropdown menu?

Reply:
^ Now I see that you use useEffect in other places. But see my comments on those useEffect hooks below